File: apps/api/app/schemas/social_link.py
"""Social link schemas"""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict, Field


class SocialLinkBase(BaseModel):
    platform: str = Field(..., max_length=100)
    url: str = Field(..., max_length=1000)


class SocialLinkCreate(SocialLinkBase):
    pass


class SocialLinkUpdate(BaseModel):
    platform: Optional[str] = Field(None, max_length=100)
    url: Optional[str] = Field(None, max_length=1000)


class SocialLinkResponse(SocialLinkBase):
    id: str
    user_id: str
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: apps/web/types/socialLink.ts
export interface SocialLink {
  id: string;
  user_id: string;
  platform: string;
  url: string;
  created_at: string;
  updated_at: string;
}

export interface SocialLinkCreate {
  platform: string;
  url: string;
}

export type SocialLinkUpdate = Partial<SocialLinkCreate>;

